src/libsync/propagatedownload.h

```diff
@@ -15,33 +15,131 @@
 
 #include "owncloudpropagator.h"
 #include "networkjobs.h"
+#include "propagatecommonzsync.h"
 
 #include <QBuffer>
 #include <QFile>
 
 namespace OCC {
 
+class GETJob : public AbstractNetworkJob
+{
+    Q_OBJECT
+protected:
+    QByteArray _etag;
+    time_t _lastModified;
+    QString _errorString;
+    SyncFileItem::Status _errorStatus;
+    bool _bandwidthLimited; // if _bandwidthQuota will be used
+    bool _bandwidthChoked; // if download is paused (won't read on readyRead())
+    qint64 _bandwidthQuota;
+    QPointer<BandwidthManager> _bandwidthManager;
+
+public:
+    GETJob(AccountPtr account, const QString &path, QObject *parent = 0) :
+        AbstractNetworkJob(account, path, parent)
+        , _etag(NULL)
```

Comment:
QByteArray is default constructed.  and NULL is specific to C.

Please actually initialize everything inline. 
The old code did not do that because we only dropped compiler that did not support C++11 very recently. But new code should use the C++11 way.